### PR TITLE
DPC-162: Add patient and provider details to attribution engine

### PR DESCRIPTION
**Pull request by nickrobison-usds**
The PR modifies the roster engine in order to add necessary details for patients and providers.

Per the discussion in DPC-161, we now require additional fields to be present in the FHIR resources. DPC-171 and DPC-172 have been created to track those issues.

The major changes is that we now require submitting actual FHIR resources to the Roster resource. This is partially implemented in that it is possible to query for an attribution relationship using only the provider NPI and beneficiary ID, but that will be resolved in DPC-90.

There are a number of things still left to do, but this provides the foundation for DPC-21, DPC-22 and DPC-90.

### CODE REVIEW

File: dpc-attribution/src/main/java/gov/cms/dpc/attribution/models/PatientEntity.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        this.dob = dob;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        PatientEntity that = (PatientEntity) o;
+        return patientID.equals(that.patientID) &&
+                beneficiaryID.equals(that.beneficiaryID) &&
```
Comment (isears-cms) [inline dpc-attribution/src/main/java/gov/cms/dpc/attribution/models/PatientEntity.java]: Thinking of a weird case where the same patient is registered under two different beneficiaries: would it be helpful for this equivalence function to identify those two patients as the same?

File: dpc-attribution/src/main/java/gov/cms/dpc/attribution/models/PatientEntity.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+    public LocalDate getDob() {
+        return dob;
+    }
+
+    public void setDob(LocalDate dob) {
+        this.dob = dob;
+    }
+
+    @Override
```
[the remark this answers is not in the document]
Comment (isears-cms) [reply dpc-attribution/src/main/java/gov/cms/dpc/attribution/models/PatientEntity.java]: I don't necessarily agree with the "cognitive complexity" here, but would it be helpful to do unit tests for some of these equivalency or fromFHIR functions?

File: dpc-attribution/src/main/java/gov/cms/dpc/attribution/models/PatientEntity.java (hunk 1 above)
Comment (nickrobison-usds) [reply dpc-attribution/src/main/java/gov/cms/dpc/attribution/models/PatientEntity.java]: That's a good point. I think bene ids are unique, but I'll have to check.

Comment (nickrobison-usds) [reply dpc-attribution/src/main/java/gov/cms/dpc/attribution/models/PatientEntity.java]: Created DPC-175 to track this. But I think we can go ahead and merge it.

**Pull request merged**